### PR TITLE
Add a dedicated const_fn gate 

### PR DESCRIPTION
**Pull request by Dylan-DPC**
**I'm submitting a ...**
  - [ ] bug fix
  - [x] feature enhancement
  - [ ] deprecation or removal
  - [ ] refactor

# Description

Uses a dedicated `const` gate instead of `nightly` 

# Motivation

these features can be easily removed when changes are stable 

# Tests

no changes required

# Related Issue(s)
#226

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -104,16 +104,13 @@
...
-    html_logo_url = "https://www.rust-lang.org/logos/rust-logo-128x128-blk-v2.png",
-    html_favicon_url = "https://www.rust-lang.org/favicon.ico",
-    html_root_url = "https://docs.rs/uuid"
-)]
+#![doc(html_logo_url = "https://www.rust-lang.org/logos/rust-logo-128x128-blk-v2.png",
+       html_favicon_url = "https://www.rust-lang.org/favicon.ico",
+       html_root_url = "https://docs.rs/uuid")]
 #![deny(warnings)]
 #![cfg_attr(not(feature = "std"), no_std)]
 #![cfg_attr(feature = "nightly", feature(const_fn))]
```
Comment (KodrAus) [inline src/lib.rs]: So we'll follow the same pattern as `serde` that uses a single blanket `unstable` feature for all `#![feature]` attributes?

Comment (Dylan-DPC) [reply src/lib.rs]: for what exactly @KodrAus ?

File: Cargo.toml [hunk 2]
```diff
@@ -43,3 +43,4 @@ v1 = []
 v3 = ["md5", "rand"]
 v4 = ["rand"]
 v5 = ["sha1", "rand"]
+const = ["nightly"]
```
Comment (kinggoesgaming) [inline Cargo.toml]: I would prefer this to be named `const_fn` as `const` could be constants or `const_fn`

File: src/core_support.rs [hunk 3]
```diff
@@ -62,7 +62,7 @@ mod tests {
     use test_util;
 
     macro_rules! check {
-        ($buf:ident, $format:expr, $target:expr, $len:expr, $cond:expr) => {
+        ($buf: ident, $format: expr, $target: expr, $len: expr, $cond: expr) => {
```
Comment (kinggoesgaming) [inline src/core_support.rs]: I think you used wrong version of `rustfmt`

**Comment by Dylan-DPC** [comment]
Git is messing things up. Will push a new PR

**Pull request closed without merging**